src/hooks/tradingView/useOhlcCandles.ts

```diff
@@ -0,0 +1,34 @@
+import { useEffect, useState } from 'react';
+
+/**
+ * @description Hook to handle drawing candles with OHLC or orderbook price
+ */
+
+export const useOhlcCandles = ({
+  ohlcToggle,
+  isChartReady,
+}: {
+  ohlcToggle: HTMLElement | null;
+  isChartReady: boolean;
+}) => {
+  const [useOhlc, setUseOhlc] = useState(true);
+
+  useEffect(() => {
+    // Initialize onClick for ohlc toggle
+    if (isChartReady && ohlcToggle) {
+      ohlcToggle.onclick = () => setUseOhlc((prev) => !prev);
+    }
+  }, [isChartReady, ohlcToggle]);
+
+  useEffect(
+    // Update ohlc button on toggle
+    () => {
+      if (useOhlc) {
+        ohlcToggle?.classList?.add('ohlc-active');
+      } else {
+        ohlcToggle?.classList?.remove('ohlc-active');
+      }
+    },
+    [useOhlc, ohlcToggle?.classList]
```

Comment:
nit -- actually apply to both this and useChartlines

would it make sense to actually include this logic in the onclick function?

so something like below, and then we don't need another useEffect. unless the classList can be updated by other things?
```
ohlcToggle.onclick = () =>  {
    setUseOhlc((prev) => {
        const isActive = !prev;
        if (isActive) ohlcToggle?.classList?.add('ohlc-active');
        else...
        setUseOhlc(isActive)

Reply:
hmm I think the only case we'd be missing is the initial render (since we default to having it on!). I think I could get around this by having a `useEffect` run just on mount to `orderLineToggle?.classList?.add('order-lines-active');` but not sure that's much cleaner 🤔 your thoughts?

Reply:
update, gonna go ahead and merge, but happy to circle back if ya feel strongly here!